scripts/firedrake-check

```diff
@@ -78,7 +78,11 @@ def run_tests(tests: tuple[str], nprocs: int, mpiexec: str) -> None:
     else:
         check_flags = "--quiet -p no:cacheprovider"
 
-    cmd = f"{mpiexec} {nprocs} {sys.executable} -m pytest {check_flags} -m parallel[{nprocs}]".split()
+    # Do not use mpiexec -n 1 as this can hang
+    if nprocs == 1:
+        cmd = f"{sys.executable} -m pytest {check_flags} -m parallel[1]".split()
+    else:
+        cmd = f"{mpiexec} {nprocs} {sys.executable} -m pytest {check_flags} -m parallel[{nprocs}]".split()
```

Comment:
1. Why not use `firedrake-run-split-tests` here?
2. If that's not possible, then make it obvious that the only difference is whether `mpiexec` is used or not.

```suggestion
    cmd = f"{sys.executable} -m pytest {check_flags} -m parallel[{nprocs}]".split()                                         
    # Only use mpiexec for parallel tests because mpiexec -n 1 can hang
    if nprocs > 1:
        mpi_cmd = f"{mpiexec} {nprocs}".split()
        cmd = mpi_cmd + cmd
```